### PR TITLE
conform to es-shim api contract

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,50 +1,57 @@
...
-	},
-	"scripts": {
-		"test": "node tests/*.js",
-		"cover": "covert tests/*.js"
-	},
-	"devDependencies": {
-		"covert": "^1.1.0",
-		"tape": "^4.2.2"
-	}
+  "name": "array.from",
```
Comment (ljharb) [inline package.json]: please preserve the indentation in this file

File: polyfill.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+module.exports = function getPolyfill() {
+	var implemented = 'from' in Array && (function () {
```
Comment (ljharb) [inline polyfill.js]: instead of all these `in` checks, would it be better to test that it's callable?

**Comment by JakeChampion** [comment]
@ljharb Fixed the indentation in package.json -- I made it all tabs as it was a mix of spaces and tabs.
I've replaced the property existence checks `'from' in Array` with callable checks `ES.IsCallable(Array.from)`.

File: implementation.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+var ES = require('es-abstract/es6');
+
+/*! https://mths.be/array-from v0.2.0 by @mathias */
+module.exports = function from(arrayLike) {
+	var defineProperty = (function() {
```
Comment (ljharb) [inline implementation.js]: all this defineProperty code can be removed if `define-properties` is added as a dep? I feel like unless we're going for "all inline, zero deps", we should use external deps whenever possible.

Comment (JakeChampion) [inline implementation.js]: I believe define-properties only does non-enumerable

Comment (ljharb) [inline implementation.js]: aha, that's true. we could use it to detect support, at least, with the `supportsDescriptors` property

**Comment by JakeChampion** [comment]
@ljharb Happy with the state of this PR?

**Comment by ljharb** [comment]
This LGTM but we should update the README to indicate that it's no longer directly usable in a `<script>` tag (which is fine because it's 2016)

**Comment by JakeChampion** [comment]
@ljharb Updated the readme, I based it off of your [array-includes readme](https://github.com/ljharb/array-includes/blob/master/README.md)

File: README.md [hunk 4]
```diff
@@ -1,31 +1,60 @@
...
 
 ```js
-require('array.from');
+var from = require('array.from');
+var assert = require('assert');
+/* when Array#from is not present */
+delete Array.from;
+var shimmedFrom = from.shim();
+assert.equal(shimmedfrom, from.getPolyfill());
+assert.deepEqual(arr.includes('foo', 1), includes(arr, 'foo', 1));
```
Comment (ljharb) [inline README.md]: we should replace this with an `Array.from` example

File: README.md [hunk 5]
```diff
@@ -1,31 +1,60 @@
...
+```
 
-// On Windows and on Mac systems with default settings, case doesn’t matter,
-// which allows you to do this instead:
-require('Array.from');
+```js
+var includes = require('array-includes');
```
Comment (ljharb) [inline README.md]: also this

**Comment by JakeChampion** [comment]
Fixed

File: README.md (hunk 5 above)
Comment (JakeChampion) [inline README.md]: copy and pasting is a dangerous thing to be doing :-/

Fixed in latest commit

**Comment by ljharb** [comment]
If you could rebase this down to a minimal set of atomic commits, this LGTM!

**Comment by JakeChampion** [comment]
Rebased

**Comment by ljharb** [comment]
Released as v1.0.0

**Pull request merged**